types/meteor/meteor.d.ts

```diff
@@ -248,7 +248,8 @@ declare module 'meteor/meteor' {
          * @param func A function that takes a callback as its final parameter
          * @param context Optional `this` object against which the original function will be invoked
          */
-        function wrapAsync(func: Function, context?: Object): any;
+        function wrapAsync<T extends Function>(func: T, context: ThisParameterType<T>): Function;
+        function wrapAsync(func: Function): Function;
```

Comment:
The first line should be enough; maybe with making the `context` optional. _(It should remove the last callback argument, but this is already better than before.)_

Reply:
Ah yes, that's fair, I'll fix that. I originally tried to write a version that did drop the callback argument, but the semantics here are messy enough that I couldn't get it to work.

(It's not just that you drop the callback argument - the callback argument becomes optional, and the return type of the resulting function changes based on whether or not the callback is provided)